File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
         unordered_map<int,int> mp;

        for(int i = 0; i < nums.size(); i++){
            int x = nums[i];
            int y = target - x;

            if(mp.count(y)){
                return {mp[y], i};  // found pair
            }

            mp[x] = i;
        }

        return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
       HashMap<Integer,Integer> mp=new HashMap<>();

        for(int i = 0; i < n; i++){
            int x = nums[i];
            int y = target - x;

            if(mp.containsKey(y)){
                return new int[]{mp.get(y), i};  // found pair
            }

            mp.put(nums[i],i);
        }

        
        return new int[]{};
    }
}